packages/cron-job/CronJobProcessor.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Draw\Component\CronJob;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Draw\Component\CronJob\Entity\CronJob;
+use Draw\Component\CronJob\Entity\CronJobExecution;
+use Draw\Component\CronJob\Message\ExecuteCronJobMessage;
+use Draw\Contracts\Process\ProcessFactoryInterface;
+use Monolog\Formatter\JsonFormatter;
+use Monolog\Level;
+use Monolog\LogRecord;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+class CronJobProcessor
+{
+    public function __construct(
+        private ManagerRegistry $managerRegistry,
+        private ProcessFactoryInterface $processFactory,
+        private MessageBusInterface $messageBus,
+    ) {
+    }
+
+    public function queue(CronJob $cronJob, bool $force = false): void
+    {
+        $manager = $this->managerRegistry->getManagerForClass(CronJobExecution::class);
+
+        $manager->persist($execution = $cronJob->newExecution($force));
+        $manager->flush();
+
+        $this->messageBus->dispatch(new ExecuteCronJobMessage($execution));
+    }
+
+    public function process(CronJobExecution $execution): void
+    {
+        $manager = $this->managerRegistry->getManagerForClass(CronJobExecution::class);
+
+        $execution->start();
+        $manager->flush();
+
+        $command = $execution->getCronJob()->getCommand();
+        $process = $this->processFactory->create(
+            [$command],
```

Comment:
Not sure it will work if command has parameter you might not be able to use the process factory...

Reply:
Did you test this with a command that have argument almosts 99 % sure it will not work